generator/generator_tests/osm_type_test.cpp

```diff
@@ -1210,8 +1208,8 @@ UNIT_CLASS_TEST(TestWithClassificator, OsmType_ReuseTags)
 
     auto const params = GetFeatureBuilderParams(tags);
 
-    TEST_EQUAL(params.m_types.size(), 2, (params));
-    TEST(params.IsTypeExist(GetType({"amenity", "parking", "private"})), (params));
+    TEST_EQUAL(params.m_types.size(), 1, (params));
+    // Better than *-private just because of lexicigraphical compare.
     TEST(params.IsTypeExist(GetType({"amenity", "parking", "fee"})), (params));
```

Comment:
a private paid parking is not a valid case? did you check its insignificant enough to be ignored?

also relying on lexicographic order doesn't seem desirable or reliable;
if we had to choose between two options then its better to pick the most important one, not the one that lexicographically is earlier

Reply:
and e.g. we might have `highway-path-bridge` and `highway-path-mountain_hiking` in one feature, right?
my understanding is that with the suggested logic only `highway-path-bridge` will stay?
but `highway-path-alpine_hiking` will replace the bridge type, right?

similar with e.g. `highway-service-bridge/busway`, `highway-track-bridge/grade2/no-access` maybe there are more examples

Reply:
Good point. Yes, basically the logic is that we keep only one unique 2 level x-y-* type for the feature. Or add exceptions like with charging_station. There is no ideal solution, but taking into account all pros/cons and tradeoffs, this logic is better than the current (before PR).

Now concrete about this examples:
- Different _hiking_ types and _grades_ should be taken out as _tags_, like current hwtag and future fee=*, access=*, because they can be applied not only to paths. https://github.com/organicmaps/organicmaps/issues/1312
- bridge/tunnel should be leaves for any type e.g. highway-service-busway-bridge Or we can consider (don't know if it is will be better for styles) also treat bridge/tunnel like tags.